lib/HdtDocument.cc

```diff
@@ -516,8 +517,9 @@ class FetchDistinctTermsWorker : public Nan::AsyncWorker {
       while (distinctTerms.size() < limit && terms->hasNext()) {
         const char* predicate = reinterpret_cast<char*>(terms->next());
 
-        // Check whether a triple with this predicate and object exists
-        hdt::IteratorTripleString *it = document->GetHDT()->search("", predicate, object.c_str());
+        // Check whether a triple with this predicate and subject or object exists
+        auto it = (!subject.empty()) ? document->GetHDT()->search(subject.c_str(), predicate, "")
```

Comment:
Can you remove `auto` and the parentheses around the condition?
Perhaps invert the condition so we don't need the `!`?

Reply:
Ternary is redundant in the end. There is no need to distinguish exclusively the case of subject or object being empty. Can pass both with their variable names and not the empty string "".